src/NuGet.Clients/PackageManagement.VisualStudio/EnvDTEProjectAdapter.cs

```diff
@@ -161,7 +167,7 @@ public bool IsLegacyCSProjPackageReferenceProject
                             _isLegacyCSProjPackageReferenceProject = true;
                         }
                     }
-                    else if (_context.NuGetProjectStyle.Equals(ProjectStyle.PackageReference.ToString(), StringComparison.OrdinalIgnoreCase))
+                    else if (restoreProjectStyle.Equals(ProjectStyle.PackageReference.ToString(), StringComparison.OrdinalIgnoreCase))
```

Comment:
discussed offline, this is more of a question of if the project system can have PackageReference.  nuget.exe won't do these extra checks so relying on the property would be best. This should stay as-is.